### ISSUE TITLE
Add Progress Bar for Simulation.simulate

### ISSUE THREAD

**Issue by jonstrutz11** [Heuro-labs/cayenne#131]
It would be nice to have a progress bar displayed during simulations (especially in Jupyter Notebook).

However, since simulations end at either a specific time or iteration, maybe you would need two separate progress bars? One for time, and one for # iterations?

What do you think?

Maybe the [tqdm](https://github.com/tqdm/tqdm) library could be useful.

**Comment by sriki18** [comment]
Oh wow, that's much longer than I expected. It must be a pretty big system. Or maybe you are running it for a longer time? Also which algorithm are you using? 

That's cool actually, likely shows that it can be done. We will take a look at it when we find some time

**Comment by jonstrutz11** [comment]
Yes it is quite a large system (100,000+ molecules) but with only a few unique species (~10). Because I'm interested in using KMC to generate many (1000s) of polymer sequences I need to start with many molecules in the system. Ideally, more than I am currently. I run it for 3600 s just using the direct algorithm. I tried Tau adaptive and didn't see much improvement in speed :/
